internal/rbd/rbd_attach.go

```diff
@@ -265,12 +263,46 @@ func appendDeviceTypeAndOptions(cmdArgs []string, isNbd, isThick bool, userOptio
 		if !strings.Contains(userOptions, setNbdIOTimeout) {
 			cmdArgs = append(cmdArgs, "--options", fmt.Sprintf("%s=%d", setNbdIOTimeout, defaultNbdIOTimeout))
 		}
+
+		if isThick {
+			// When an image is thick-provisioned, any discard/unmap/trim
+			// requests should not free extents.
+			cmdArgs = append(cmdArgs, "--options", "notrim")
+		}
+	}
+
+	if userOptions != "" {
+		// userOptions is appended after, possibly overriding the above
+		// default options.
+		cmdArgs = append(cmdArgs, "--options", userOptions)
+	}
+
+	return cmdArgs
+}
+
+func appendKRbdDeviceTypeAndOptions(cmdArgs []string, isThick bool, userOptions string) []string {
+	cmdArgs = append(cmdArgs, "--device-type", accessTypeKRbd)
+
+	isMap := true
+	for _, item := range cmdArgs {
+		if item == "unmap" {
+			isMap = false
+		}
```

Comment:
Move this to a generic helper function to utils folder. which can be used later.

```suggestion
func CheckOptionContains(options []string, opt string) bool {
	for _, o := range options {
		if o == opt {
			return true
		}
	}

	return false
}
```

Reply:
Done now!